src/main/java/org/threadly/litesockets/udp/UDPClient.java

```diff
@@ -275,45 +119,55 @@ public int getTimeout() {
   }
 
   @Override
-  protected void setConnectionStatus(Throwable t) {
+  protected void setConnectionStatus(final Throwable t) {
+    //UDP has no "connection"
   }
-  
+
   @Override
-  public SocketAddress getRemoteSocketAddress() {
+  public InetSocketAddress getRemoteSocketAddress() {
     return remoteAddress;
   }
 
   @Override
-  public SocketAddress getLocalSocketAddress() {
-    if(this.udpServer.channel != null) {
-      return udpServer.channel.socket().getLocalSocketAddress();
+  public InetSocketAddress getLocalSocketAddress() {
+    if(udpServer.getSelectableChannel() != null) {
+      return (InetSocketAddress)udpServer.getSelectableChannel().socket().getLocalSocketAddress();
     }
     return null;
   }
-  
+
   @Override
   public String toString() {
     return "UDPClient:FROM:"+getLocalSocketAddress()+":TO:"+getRemoteSocketAddress();
   }
 
-  /**
-   * Implementation of the SimpleByteStats.
-   */
-  private static class ClientByteStats extends SimpleByteStats {
-    public ClientByteStats() {
-      super();
+  @Override
+  public ListenableFuture<?> write(final ByteBuffer bb) {
+    addWriteStats(bb.remaining());
+    if(!closed.get()) {
+      try {
+        udpServer.channel.send(bb, remoteAddress);
+      } catch (IOException e) {
+      }
     }
+    return FutureUtils.immediateResultFuture(true);
```

Comment:
This seems like it would be called frequently enough that it would be worth just constructing a static instance of ImmediateResultListenableFuture ( http://threadly.github.io/threadly/javadocs/4.3.0/org/threadly/concurrent/future/ImmediateResultListenableFuture.html )...that way we are not GC'ing a ton of these.